lib/evmone/tracing.cpp

```diff
@@ -111,38 +109,31 @@ class InstructionTracer : public Tracer
         const auto& ctx = m_contexts.top();
 
         const auto opcode = ctx.code[pc];
-        m_out << "{";
-        m_out << R"("pc":)" << std::dec << pc;
-        m_out << R"(,"op":)" << std::dec << int{opcode};
-        m_out << R"(,"opName":")" << get_name(opcode) << '"';
-        m_out << R"(,"gas":0x)" << std::hex << gas;
-        output_stack(stack_top, stack_height);
+        m_buf << "{";
+        m_buf << R"("pc":)" << std::dec << pc;
+        m_buf << R"(,"op":)" << std::dec << int{opcode};
+        m_buf << R"(,"gas":"0x)" << std::hex << gas << '"';
+        m_buf << R"(,"gasCost":"0x)" << std::hex << instr::gas_costs[state.rev][opcode] << '"';
 
         // Full memory can be dumped as evmc::hex({state.memory.data(), state.memory.size()}),
         // but this should not be done by default. Adding --tracing=+memory option would be nice.
-        m_out << R"(,"memorySize":)" << std::dec << state.memory.size();
-
-        m_out << "}\n";
-    }
+        m_buf << R"(,"memSize":)" << std::dec << state.memory.size();
 
-    void on_execution_end(const evmc_result& result) noexcept override
-    {
-        const auto& ctx = m_contexts.top();
+        output_stack(stack_top, stack_height);
+        if (!state.return_data.empty())
+            m_buf << R"("returnData":"0x)" << evmc::hex(state.return_data) << '"';
```

Comment:
```suggestion
            m_buf << R"(,"returnData":"0x)" << evmc::hex(state.return_data) << '"';
```
This line is not covered by tests and in the same time there is a bug in JSON syntax.